src/test/ui/consts/miri_unleashed/mutable_const.rs

```diff
@@ -2,7 +2,7 @@
 
 #![feature(const_raw_ptr_deref)]
 #![feature(const_mut_refs)]
-#![deny(const_err)]
+#![deny(const_err)] // FIXME: ICEs with allow! See #71316.
```

Comment:
yea, as you noted on the issue, I expect some things to ICE with unleash, and I think that's perfectly ok.

Reply:
can you add the appropriate compiletest comments to this file and allow the lint?

Reply:
I don't think it's okay that they ICE, but we can discuss that in the issue. ;)

I'll try to make the test check for an ICE instead.

Reply:
Oh, actually we already have a test for exactly that, `mutable_const2.rs`.

Reply:
@oli-obk I think my latest commit addresses... oh you already saw it and `r+`'d, never mind. ;)